app.py

```diff
@@ -31,46 +32,168 @@ def init_app(self):
+            # the parent folder of the path is valid and try opening that.
+            # todo: The issue with this logic is that possibly it was a directory that just didn't exist,
+            # so we would just be gathering the next directory up,
+            # ideally need a better way to handle sequences.
+            parent_dir = os.path.dirname(path)
+            self._launch_filemanager_for_folder(parent_dir)
+
+            
+    def _launch_filemanager_for_folder(self, path):
+        """
+        This method will take a path to a folder and open it in the OS's default file manager.
+        :param path: A folder path
+        :raises: Exception: if the Platform is not supported
+        :raises: ValueError: if the path is not a valid directory.
+        """
+        self.log_debug("Launching file system viewer for folder %s" % path)
+
+        # Check that we don't have a file path.
+        if not os.path.isdir(path):
+            raise ValueError(
+                "The path \"%s\" is not a valid directory." % path)
+
+        # get the setting
         system = sys.platform
-        
-        # run the app
-        if system == "linux2":
-            cmd = 'xdg-open "%s"' % path
+        self.log_debug("Detected system: %s" % system)
+
+        # build the commands for opening the folder on the various OS's
+        if system.startswith("linux"):
+            cmd_args = ["xdg-open", path]
         elif system == "darwin":
-            cmd = 'open "%s"' % path
+            cmd_args = ["open", path]
         elif system == "win32":
-            cmd = 'cmd.exe /C start "Folder" "%s"' % path
+            cmd_args= ["cmd.exe", "/C", "start", path ]
         else:
             raise Exception("Platform '%s' is not supported." % system)
-        
-        self.log_debug("Executing command '%s'" % cmd)
-        exit_code = os.system(cmd)
+
+        self.log_debug("Executing command '%s'" % cmd_args)
+        exit_code = subprocess.call(cmd_args)
         if exit_code != 0:
-            self.log_error("Failed to launch '%s'!" % cmd)
+            self.log_error("Failed to launch '%s'!" % cmd_args)
+
+    def _launch_filemanager_for_file(self, path):
+        """
+        This method will take a path to a file and open it in the OS's default file manager.
+        This is only used for files, for folders use _launch_filemanager_for_folder.
+        :param path: A file path
+        :raises: Exception: if the Platform is not supported
+        :raises: ValueError: if the path is not a valid file.
+        """
+        self.log_debug("Launching file system viewer for file %s" % path)
+
+        if not os.path.isfile(path):
+            raise ValueError(
+                "The path \"%s\" is not a valid file path." % path)
+
+        # get the setting
+        system = sys.platform
+        self.log_debug("Detected system: %s" % system)
+
+        if system.startswith("linux"):
```

Comment:
If we introduce an `open_file` function in tk-core, maybe `select_file_in_browser` wouldn't be such a bad thing either.